### PR TITLE
Add template variable subpage UI

### PR DESCRIPTION
**Pull request by AlirieGray**
Closes: https://github.com/influxdata/applications-team-issues/issues/284

This PR adds a subpage to the /organizations page for template variables. The page includes a button that opens an overlay to create a new template variable. However, this UI is not currently connected to the backend.

In addition, some types are moved out of the /v2 folder to the /types folder.

  - [x] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

File: ui/src/organizations/components/CreateVariableOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    )
+  }
+
+  private handleCreateVariable = () => {
+    const {onCloseModal} = this.props
+    onCloseModal()
+  }
+
+  private handleChangeInput = (e: ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value
```
Comment (OfTheDelmer) [inline ui/src/organizations/components/CreateVariableOverlay.tsx]:
```suggestion
    const {value, name} = e.target
    this.setState({[name]: value})
```

File: ui/src/organizations/components/CreateVariableOverlay.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+              />
+            </OverlayFooter>
+          </OverlayBody>
+        </Form>
+      </OverlayContainer>
+    )
+  }
+
+  private handleCreateVariable = () => {
+    const {onCloseModal} = this.props
```
Comment (OfTheDelmer) [inline ui/src/organizations/components/CreateVariableOverlay.tsx]:
```suggestion
this.props.onCloseModal()
```

File: ui/src/organizations/components/CreateVariableOverlay.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+              </Form.Element>
+            </div>
+
+            <Form.Element label="Value">
+              <div className="overlay-flux-editor">
+                <FluxEditor
+                  script={script}
+                  onChangeScript={this.handleChangeScript}
+                  visibility="visible"
+                  status={{text: '', type: ''}}
```
Comment (OfTheDelmer) [inline ui/src/organizations/components/CreateVariableOverlay.tsx]: Maybe there could be some kind default status prop in the `FluxEditor` so we don't have to specify this when we're not updating it

File: ui/src/organizations/components/CreateVariableOverlay.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+  public render() {
+    const {onCloseModal} = this.props
+    const {nameInputStatus, name, script} = this.state
+
+    return (
+      <OverlayContainer maxWidth={1000}>
+        <OverlayHeading
+          title="Create Variable"
+          onDismiss={this.props.onCloseModal}
```
Comment (OfTheDelmer) [inline ui/src/organizations/components/CreateVariableOverlay.tsx]:
```suggestion
          onDismiss={onCloseModal}
```

File: ui/src/organizations/components/Variables.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import {OverlayState} from 'src/types'
+
+interface Props {}
+
+interface State {
+  searchTerm: string
+  overlayState: OverlayState
+}
+
+export default class Buckets extends PureComponent<Props, State> {
```
Comment (OfTheDelmer) [inline ui/src/organizations/components/Variables.tsx]:
```suggestion
export default class Variables extends PureComponent<Props, State> {
```

File: ui/src/types/index.ts [hunk 6]
```diff
@@ -38,12 +38,14 @@ import {
...
 import {FluxTable, ScriptStatus, SchemaFilter, RemoteDataState} from './flux'
 import {WriteDataMode} from './dataExplorer'
+import {OverlayState} from 'src/types/overlay'
 
 export {
   TemplateType,
   TemplateValue,
   TemplateValueType,
   Template,
+  OverlayState,
```
Comment (OfTheDelmer) [inline ui/src/types/index.ts]: nice

File: ui/src/organizations/components/CreateVariableOverlay.tsx (hunk 3 above)
Comment (AlirieGray) [reply ui/src/organizations/components/CreateVariableOverlay.tsx]: good idea 👍

File: ui/src/organizations/components/Variables.tsx (hunk 5 above)
Comment (AlirieGray) [reply ui/src/organizations/components/Variables.tsx]: 😱 oops

**Pull request merged**